pkg/cloudprovider/providers/azure/azure_routes.go

```diff
@@ -0,0 +1,189 @@
+package azure
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/cloudprovider"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/glog"
+)
+
+// ListRoutes lists all managed routes that belong to the specified clusterName
+func (az *AzureCloud) ListRoutes(clusterName string) (routes []*cloudprovider.Route, err error) {
+	glog.Infof("list: START clusterName=%q", clusterName)
+	glog.Infof("list: getting the route table. routeTableName=%q", az.RouteTableName)
+	routeTable, err := az.RouteTablesClient.Get(az.ResourceGroup, az.RouteTableName, "")
+	if existsRouteTable, err := checkResourceExistsFromError(err); err != nil {
+		return nil, err
+	} else if !existsRouteTable {
+		glog.Infof("list: routing table didn't exist. routeTableName=%q", az.RouteTableName)
+		return []*cloudprovider.Route{}, nil
+	}
+
+	var kubeRoutes []*cloudprovider.Route
+	if routeTable.Properties.Routes != nil {
+		kubeRoutes = make([]*cloudprovider.Route, len(*routeTable.Properties.Routes))
+		for i, route := range *routeTable.Properties.Routes {
+			instance := getInstanceName(*route.Name)
+			cidr := *route.Properties.AddressPrefix
+			glog.Infof("list: * instance=%q, cidr=%q", instance, cidr)
+
+			kubeRoutes[i] = &cloudprovider.Route{
+				Name:            *route.Name,
+				TargetInstance:  instance,
+				DestinationCIDR: cidr,
+			}
+		}
+	} else {
+		kubeRoutes = []*cloudprovider.Route{}
+	}
+
+	glog.Info("list: FINISH")
+	return kubeRoutes, nil
+}
+
+// CreateRoute creates the described managed route
+// route.Name will be ignored, although the cloud-provider may use nameHint
+// to create a more user-meaningful name.
+func (az *AzureCloud) CreateRoute(clusterName string, nameHint string, kubeRoute *cloudprovider.Route) error {
+	glog.Infof("create: clusterName=%q instance=%q cidr=%q", clusterName, kubeRoute.TargetInstance, kubeRoute.DestinationCIDR)
+
+	glog.Infof("create: getting the routetable. routeTableName=%q", az.RouteTableName)
+	routeTable, err := az.RouteTablesClient.Get(az.ResourceGroup, az.RouteTableName, "")
+	if existsRouteTable, err := checkResourceExistsFromError(err); err != nil {
+		return err
+	} else if !existsRouteTable {
+		glog.Infof("create: routetable needs creation. routeTableName=%q", az.RouteTableName)
+		routeTable = network.RouteTable{
+			Name:       to.StringPtr(az.RouteTableName),
+			Location:   to.StringPtr(az.Location),
+			Properties: &network.RouteTablePropertiesFormat{},
+		}
+
+		_, err = az.RouteTablesClient.CreateOrUpdate(az.ResourceGroup, az.RouteTableName, routeTable, nil)
```

Comment:
probably going to punt on this, shorthand form doesn't save any lines unless I have an even longer (too long, imo) line here.